File: src/com/charter/enterprise/collatz/model/CollatzDataModel.java
package com.charter.enterprise.collatz.model;

public class CollatzDataModel implements Comparable<CollatzDataModel> {

    private long collatzNumber;

    private long collatzIterations;

    public CollatzDataModel(long collatzNumber, long collatzIterations) {
        this.collatzNumber = collatzNumber;
        this.collatzIterations = collatzIterations;
    }

    public CollatzDataModel(CollatzDataModel collatzDataModel) {
        this.collatzNumber = collatzDataModel.getCollatzNumber();
        this.collatzIterations = collatzDataModel.getCollatzIterations();
    }


    @Override
    public int compareTo(CollatzDataModel collatzDataModel) {
        return Long.compare(this.collatzNumber, collatzDataModel.getCollatzNumber());
    }

    public long getCollatzNumber() {
        return collatzNumber;
    }

    public void setCollatzNumber(long collatzNumber) {
        this.collatzNumber = collatzNumber;
    }

    public long getCollatzIterations() {
        return collatzIterations;
    }

    public void setCollatzIterations(long collatzIterations) {
        this.collatzIterations = collatzIterations;
    }
}

File: src/model/collatz-data-model.ts
export class CollatzDataModel {
    private collatzNumber: number;
    private collatzIterations: number;

    constructor(collatzNumber: number, collatzIterations: number) {
        this.collatzNumber = collatzNumber;
        this.collatzIterations = collatzIterations;
    }

    static fromModel(model: CollatzDataModel): CollatzDataModel {
        return new CollatzDataModel(model.getCollatzNumber(), model.getCollatzIterations());
    }

    compareTo(other: CollatzDataModel): number {
        return this.collatzNumber - other.collatzNumber;
    }

    getCollatzNumber(): number {
        return this.collatzNumber;
    }

    setCollatzNumber(collatzNumber: number): void {
        this.collatzNumber = collatzNumber;
    }

    getCollatzIterations(): number {
        return this.collatzIterations;
    }

    setCollatzIterations(collatzIterations: number): void {
        this.collatzIterations = collatzIterations;
    }
}
